app/internal/security/dependancies.py

```diff
@@ -0,0 +1,66 @@
+from typing import Union
+
+from app.dependencies import get_db
+from app.database.models import User
+from app.internal.security.ouath2 import (
+    Depends, Session, check_jwt_token,
+    get_authorization_cookie, HTTP_401_UNAUTHORIZED,
+    HTTPException)
+from starlette.requests import Request
+
+
+async def get_user(
+        db: Session, user_id: int, username: str, path: str) -> User:
+    '''
+    Helper for dependency functions.
+    Recives user details from decrypted jwt token,
+    and check them against the database.
+    returns User base model instance if succeeded,
+    raises HTTPException if fails.
+    '''
+    db_user = await User.get_by_username(db, username=username)
+    if db_user and db_user.id == user_id:
+        return db_user
+    else:
+        raise HTTPException(
```

Comment:
Prefer not raising it as an http exception - let the route do it.

In these function you need to return a Pythonic response - which should probably be None, in this case.

Reply:
![image](https://user-images.githubusercontent.com/71118404/107736404-a3caa780-6d0a-11eb-8747-e5f197bf7e86.png)
The same goes for both of the comments:
The whole point of these functions, is to protect a certain route from not logged in user be able to reach it.
In any case which that i don't raise an Exception, but returning a  Boolean, that will not do the job. I already checked that, and This is precisely the case with the 'optional_logged_in_user' dependency that i made, which is not protecting the route.
This is why i configured 'auth_exception_handler' the way i did in the first place:
So i can create a dependency to protect a route, and at the same time render a specific message about what failed, and 'remembering' the user's original request